lib/pbench/server/cache_manager.py

```diff
@@ -107,28 +107,31 @@ def __init__(self, dir_name: str, path: Path):
         else:
             self.location = Path(dir_name) / str(path).split(f"{dir_name}/")[1]
 
-        try:
-            resolve_path = path.resolve(strict=True)
-        except FileNotFoundError:
-            raise FileNotFoundError(f"The '{path}' does not resolve to a real location")
-
-        if resolve_path.name == dir_name:
-            self.resolve_path = dir_name
-        else:
-            self.resolve_path = (
-                Path(dir_name) / str(resolve_path).split(f"{dir_name}/")[1]
-            )
-
+        self.resolve_path = None
         self.resolve_type = None
         self.size = None
         if path.is_symlink():
             self.type = "symlink"
-            self.resolve_type = "directory" if resolve_path.is_dir() else "file"
+            try:
+                resolve_path = path.resolve(strict=True)
+                self.resolve_type = "directory" if resolve_path.is_dir() else "file"
+            except FileNotFoundError:
+                resolve_path = path.readlink()
+                self.resolve_type = "symlink"
```

Comment:
If the link can be resolved, it will resolve to either a file or a directory (well...or to an "other" which the code currently considers to be a "file" which might be the least of evils...); if it cannot be _fully_ resolved, then, well, either it's pointing at a nonexistent file (in which case, we don't have a type...even "other" is presumably wrong) or it's pointing at another symlink (which is broken), so either way, it's a symlink....  Whether we should try to resolve it to the path of the broken link or whether we should leave the resolution of the initial link is...well...does anyone really care??  🤷

Reply:
Maybe we can keep this stored as readlink for now and take care of it at the time when we will be resolving symlinks.

Reply:
If we are going to keep `resolve_type` (as you seem inclined to do 😉), then setting `resolve_type` to `"symlink"` here is fine (in my opinion), and setting `resolve_path` to the result of `readlink()` is perfectly adequate, at least for now.